File: src/com/bivc/cimsmgs/db/PrintDataPhrase.java
package com.bivc.cimsmgs.db;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * @author p.dzeviarylin
 */

@JsonIgnoreProperties({"printData"})
@JsonInclude(JsonInclude.Include.NON_NULL)
public class PrintDataPhrase {
    private Long hid;
    private PrintData printData;
    private Integer sort;
    private String name;
    private String descr;
    private Byte fontSize;
    private Boolean bold;
    private Boolean uppercase;
    private Boolean border;
    private Boolean underline;
    private String fontFamily;
    private Byte leading;
    private Integer rotate;

    public Integer getRotate() {
        return rotate;
    }

    public void setRotate(Integer rotate) {
        this.rotate = rotate;
    }

    public Byte getLeading() {
        return leading;
    }

    public void setLeading(Byte leading) {
        this.leading = leading;
    }

    public String getFontFamily() {
        return fontFamily;
    }

    public void setFontFamily(String fontFamily) {
        this.fontFamily = fontFamily;
    }

    public Boolean getUnderline() {
        return underline;
    }

    public void setUnderline(Boolean underline) {
        this.underline = underline;
    }

    public Boolean getBorder() {
        return border;
    }

    public void setBorder(Boolean border) {
        this.border = border;
    }

    public Boolean getUppercase() {
        return uppercase;
    }

    public void setUppercase(Boolean uppercase) {
        this.uppercase = uppercase;
    }

    public Boolean getBold() {
        return bold;
    }

    public void setBold(Boolean bold) {
        this.bold = bold;
    }

    public Byte getFontSize() {
        return fontSize;
    }

    public void setFontSize(Byte fontSize) {
        this.fontSize = fontSize;
    }

    public String getDescr() {
        return descr;
    }

    public void setDescr(String descr) {
        this.descr = descr;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getSort() {
        return sort;
    }

    public void setSort(Integer sort) {
        this.sort = sort;
    }

    public PrintData getPrintData() {
        return printData;
    }

    public void setPrintData(PrintData printData) {
        this.printData = printData;
    }

    public Long getHid() {
        return hid;
    }

    public void setHid(Long hid) {
        this.hid = hid;
    }
}

File: WebContent/app/model/PrintDataPhrase.js
Ext.define('TK.model.PrintDataPhrase', {
    extend:'Ext.data.Model',

    requires: [
        'TK.model.PrintData'
    ],

    idProperty:'hid',
    fields:[
        'name',
        'descr',
        'fontFamily',
        {name:'fontSize', type:'int', useNull: true},
        {name:'leading', type:'int', useNull: true},
        {name:'sort', type:'int', useNull: true},
        {name:'rotate', type:'int', useNull: true},
        {name:'border', type:'boolean', defaultValue: false},
        {name:'underline', type:'boolean', defaultValue: false},
        {name:'bold', type:'boolean', defaultValue: false},
        {name:'uppercase', type:'boolean', defaultValue: false},
        {name:'hid', type:'int'}
    ],
    validations:[
        {type:'presence', field:'name'},

        {type:'length', field:'name', max:20},
        {type:'length', field:'descr', max:300},
        {type:'length', field:'fontFamily', max:30},
        {type:'format', field:'fontSize', matcher: /[0-9]{0,2}$/},
        {type:'format', field:'leading', matcher: /[0-9]{0,2}$/}
    ],
    belongsTo:[{
        model:'TK.model.PrintData',
        getterName:'getPrintData',
        setterName:'setPrintData',
        associationKey:'printData',
        primaryKey:'hid',
        foreignKey:'printData.hid'
    }]
});